rust/core-lib/src/tabs.rs

```diff
@@ -390,9 +390,15 @@ impl CoreState {
             None => return,
         };
 
-        let ed = &self.editors[&buffer_id];
+        let fin_text = if let Some(mut save_ctx) = self.make_context(view_id) {
+            save_ctx.pre_save_newline()
+        } else {
+            let editor = &self.editors[&buffer_id].borrow();
```

Comment:
If we can't make a context, the only reasonable explanation is that the view or buffer is missing, so this branch might be redundant? It's very hard to imagine a situation where we get here.

In fact: assuming we needed a `ViewId` in order to get the `BufferId` then we should be able to guarantee that make_context will work, and we can just unwrap.

Reply:
Alright